src/daemon.rs

```diff
@@ -139,84 +143,121 @@ pub fn main() {
         }
     }
 
+    let mut signals = Signals::new(&[SIGUSR1, SIGUSR2, SIGHUP, SIGINT])?;
+    let mut paused = false;
+    let mut temp_paused = false;
+
+    let mut last_hotkey: Option<config::Hotkey> = None;
+    let mut keyboard_states: Vec<KeyboardState> = Vec::new();
+    let mut keyboard_stream_map = StreamMap::new();
+    for (i, mut device) in keyboard_devices.into_iter().enumerate() {
+        device.update_auto_repeat(&AutoRepeat { delay: 0, period: 0 })?;
+        keyboard_stream_map.insert(i, device.into_event_stream()?);
+        keyboard_states.push(KeyboardState::new());
+    }
+
+    // the initial sleep duration is never read because last_hotkey is initialized to None
+    let hotkey_repeat_timer = sleep(Duration::from_millis(0));
+    tokio::pin!(hotkey_repeat_timer);
```

Comment:
Would be nice if an error message stating that said hotkey is in cooldown could be printed.

Reply:
When should this message be printed?

Reply:
> When should this message be printed?

If the same hotkey is attempted to be ran again before the time cooldown time runs out.

Reply:
Currently the cooldown only has an effect when the hotkey is held down, not when it is pressed multiple times. Should that be changed?

Reply:
> Currently the cooldown only has an effect when the hotkey is held down, not when it is pressed multiple times. Should that be changed?

Nope that sounds fine.

Reply:
what I meant was that if the hotkey is being held down while it has already run before the cooldown running out, it would be nice to print a log of it. 

For now this should be fine. I'll go ahead and merge this.